webpack.config.js

```diff
@@ -0,0 +1,69 @@
+const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
+const BundleAnalyzerPlugin = require("webpack-bundle-analyzer")
+  .BundleAnalyzerPlugin;
+
+const deepClone = a => {
+  if (!a) return a;
+
+  const b = new a.constructor();
+  for (let key in a) {
+    const val = a[key];
+    b[key] = typeof val === "object" ? deepClone(val) : val;
+  }
+  return b;
+};
+
+const config = Object.freeze({
```

Comment:
Can you explain the use of Object.freeze. I looked it up but am having some trouble understanding. If const locks the bindings so you can assign new values and freeze locks properties so you cannot assign new values, isnt it redundant to use both?

Reply:
`Object.freeze` prevents reassigning values to the object passed (just the top level). Its just a safety technique to reduce the chance of modification.

```js
const obj = Object.freeze({
  key: "value"
})

obj.key = "other value" // errors here for trying to change the object
```